### PR TITLE
wire extension settings to plugins

### PR DESCRIPTION
**Pull request by yuval-k**

### CODE REVIEW

File: projects/gloo/pkg/plugins/plugin_interface.go [hunk 1]
```diff
@@ -7,12 +7,13 @@ import (
 	envoylistener "github.com/envoyproxy/go-control-plane/envoy/api/v2/listener"
 	envoyroute "github.com/envoyproxy/go-control-plane/envoy/api/v2/route"
 	envoyhttp "github.com/envoyproxy/go-control-plane/envoy/config/filter/network/http_connection_manager/v2"
-	"github.com/solo-io/gloo/projects/gloo/pkg/api/v1"
+	v1 "github.com/solo-io/gloo/projects/gloo/pkg/api/v1"
```
Comment (ilackarms) [inline projects/gloo/pkg/plugins/plugin_interface.go]: why this change?

File: projects/gloo/pkg/syncer/setup_syncer.go [hunk 2]
```diff
@@ -218,19 +218,19 @@ func (s *setupSyncer) Setup(ctx context.Context, kubeCache kube.SharedCache, mem
...
 }
 
-func RunGloo(opts bootstrap.Opts) error {
-	return RunGlooWithExtensions(opts, Extensions{})
+func RunGloo(opts bootstrap.Opts, extensionsSettings *v1.Extensions) error {
+	return RunGlooWithExtensions(opts, Extensions{}, extensionsSettings)
 }
 
-func RunGlooWithExtensions(opts bootstrap.Opts, extensions Extensions) error {
+func RunGlooWithExtensions(opts bootstrap.Opts, extensions Extensions, extensionsSettings *v1.Extensions) error {
```
Comment (rickducott) [inline projects/gloo/pkg/syncer/setup_syncer.go]: Should we put extensionsSettings inside Extensions?

Comment (yuval-k) [reply projects/gloo/pkg/syncer/setup_syncer.go]: as discussed this is not needed

File: projects/gloo/pkg/plugins/plugin_interface.go (hunk 1 above)
Comment (yuval-k) [reply projects/gloo/pkg/plugins/plugin_interface.go]: no idea. my ide does it automatically every time i save

Comment (yuval-k) [reply projects/gloo/pkg/plugins/plugin_interface.go]: i figure it might be part of go fmt so i just went with it 😀
